### PR TITLE
607 password generator

### PR DESCRIPTION
**Pull request by Mefpef**
PR Title format: 

    Pull request title with short description (#607)

To make clear that you license your contribution under the MIT you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [MIT license](https://opensource.org/licenses/MIT)

PRs without ticked MIT license will bot be accepted - Sorry.

### CODE REVIEW

File: tracardi/process_engine/action/v1/password_generator_action.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    max_length: int
+    min_length: int
+    uppercase: int
+    lowercase: int
+    special_characters: int
+
+    @validator("min_length")
+    def check_min_max_value(cls, value, values):
+        if value > values["max_length"]:
+            raise ValueError("Minimum length is bigger than maximum length")
```
Comment (DawidekZagajnik) [inline tracardi/process_engine/action/v1/password_generator_action.py:22]: f"Minimal length `{value}` cannot be bigger than given maximal length `{values['max_length']}`"

File: tracardi/process_engine/action/v1/password_generator_action.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        start=False,
+        spec=Spec(
+            module=__name__,
+            className='PasswordGeneratorAction',
+            inputs=["payload"],
+            outputs=["password"],
+            version='0.7.2',
+            license="MIT",
+            author="Mateusz Zitaruk",
+            init={"min_length": None,
```
Comment (DawidekZagajnik) [inline tracardi/process_engine/action/v1/password_generator_action.py:56]: tab after first and last curly bracket to be more readable

**Review by DawidekZagajnik — APPROVED** [review]
pretty much perfect!

File: tracardi/process_engine/action/v1/password_generator_action.py (hunk 2 above)
Comment (atompie) [inline tracardi/process_engine/action/v1/password_generator_action.py:56]: I would set default values for these properties. This way users can quickly set up the plugin. The default values are up to you.

File: tracardi/process_engine/action/v1/password_generator_action.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+                  "max_length": None,
+                  "uppercase": None,
+                  "lowercase": None,
+                  "special_characters": None},
+            manual="password_generator_action",
+            form=Form(
+                groups=[
+                    FormGroup(
+                        name="Plugin config",
```
Comment (atompie) [inline tracardi/process_engine/action/v1/password_generator_action.py]: Give a more meaningful name. eg. Password generator settings.

File: tracardi/process_engine/action/v1/password_generator_action.py [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+                                )
+                            ),
+                            FormField(
+                                id="special_characters",
+                                name="Special characters",
+                                description="Please provide number of special characters.",
+                                component=FormComponent(
+                                    type="text",
+                                    props={
+                                        "label": "Special characters"
```
Comment (atompie) [inline tracardi/process_engine/action/v1/password_generator_action.py:120]: Labels could also be more meaningful eg. Special character number

**Review by DawidekZagajnik — APPROVED** [review]

**Review by atompie — APPROVED** [review]

**Pull request merged**